solc/CommandLineInterface.cpp

```diff
@@ -239,6 +255,22 @@ void CommandLineInterface::handleIROptimized(string const& _contractName)
 	}
 }
 
+void CommandLineInterface::handleIROptimizedAst(string const& _contractName)
+{
+	solAssert(CompilerInputModes.count(m_options.input.mode) == 1);
+
+	if (!m_options.compiler.outputs.irOptimizedAstJson)
+		return;
+
+	if (!m_options.output.dir.empty())
+		createFile(m_compiler->filesystemFriendlyName(_contractName) + "_opt_yul.json", util::jsonPrint(m_compiler->yulIROptimizedAst(_contractName), util::JsonFormat{ util::JsonFormat::Compact }));
```

Comment:
The file name should include `ast` somewhere to make it clear what the output is:

```suggestion
		createFile(m_compiler->filesystemFriendlyName(_contractName) + "_opt_yul_ast.json", util::jsonPrint(m_compiler->yulIROptimizedAst(_contractName), util::JsonFormat{ util::JsonFormat::Compact }));
```

Same for unoptimized.

By the way, looks like for Solidity ASTs the file name ends with `_json.ast`, which is weird. It's a JSON file so `_ast.json` would be more reasonable. Changing that would be breaking though so I guess we have to leave that as is. Let's at least get it right for Yul.